src/test/java/com/epam/test_generator/services/JiraServiceTest.java

```diff
@@ -0,0 +1,257 @@
+package com.epam.test_generator.services;
+
+import com.epam.test_generator.dao.impl.JiraProjectDAO;
+import com.epam.test_generator.dao.impl.JiraStoryDAO;
+import com.epam.test_generator.dao.impl.JiraSubStroryDAO;
+import com.epam.test_generator.dao.interfaces.*;
+import com.epam.test_generator.entities.*;
+import com.epam.test_generator.pojo.JiraProject;
+import com.epam.test_generator.pojo.JiraStory;
+import com.epam.test_generator.pojo.JiraSubTask;
+import net.rcarz.jiraclient.Issue;
+import net.rcarz.jiraclient.Issue.SearchResult;
+import net.rcarz.jiraclient.JiraClient;
+import net.rcarz.jiraclient.JiraException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.mockito.AdditionalMatchers.not;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class JiraServiceTest {
+    @Mock
+    private JiraClient client;
+
+    @Mock
+    private JiraSubStroryDAO jiraSubStoryDAO;
+
+    @Mock
+    private JiraProjectDAO jiraProjectDAO;
+
+    @Mock
+    private JiraStoryDAO jiraStoryDAO;
+
+    @Mock
+    private ProjectDAO projectDAO;
+
+    @Mock
+    private SuitDAO suitDAO;
+
+    @Mock
+    private CaseDAO caseDAO;
+
+    @Mock
+    private UserDAO userDAO;
+
+    @Mock
+    private RemovedIssueDAO removedIssueDAO;
+
+    @Mock
+    private JiraSubTask jiraSubTask;
+
+    @Mock
+    private User user;
+
+    @Mock
+    private Suit suit;
+
+    @Mock
+    private Project project;
+
+    @Mock
+    private JiraStory jiraStory;
```

Comment:
It is really cool that you write tests, but I think you overdone with mocking. If you look on you test when it is running, you will see NULL in big part of your fileds. But I think that when you run test you must see what contains in your fields. You can create a real JiraStory, set values into fields and it is really help you when you will be execute you test, because you will see how data from your fields will be move and transforme. Also you can create a real JiraProject, JiraSubTask, Project, User, Suit, Case. If you have a simple pojo that you can create, I think it must be created. You must mock only that things which you really difficult to create. 
And this applies to all tests.

Reply:
Valuable comment! Thanks!